addons/payment_paypal/models/paypal_s2s.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-'8' "-*-"
+
+import base64
+import json
+import logging
+import urllib
+import urllib2
+
+from odoo import api, models, _
+from odoo.tools.safe_eval import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+class AcquirerPaypalS2s(models.Model):
```

Comment:
My bad, you implemented this function but for the wrong reasons.

The `paypal_s2s_form_validate` function is called by the `s2s_process` which calls it to check if all mandatory field of the s2s form (the one where you put your card information) are filled in (and valid).